akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowSpec.scala

```diff
@@ -25,7 +28,11 @@ import akka.stream.impl.fusing.GraphInterpreterShell
 object FlowSpec {
   class Fruit
   class Apple extends Fruit
-  val apples = () ⇒ Iterator.continually(new Apple)
+  class Orange extends Fruit
+  val fruits = () ⇒ new Iterator[Fruit] {
+    override def hasNext: Boolean = true
+    override def next(): Fruit = if (ThreadLocalRandom.current().nextBoolean()) new Apple else new Orange
+  }
```

Comment:
Was the intention of this modification so that `FlowSpec.fruits` could be used on `collectType` tests?

Reply:
Mostly wanted the fruits collection in:

```
"be covariant" in {
      val f1: Source[Fruit, _] = Source.fromIterator[Fruit](fruits)
      val p1: Publisher[Fruit] = Source.fromIterator[Fruit](fruits).runWith(Sink.asPublisher(false))
```

be a real fruits collection